### DISCUSSION TITLE
ESP-C3-32S 4MB which firmware to use?

### DISCUSSION THREAD

**Discussion by redpixy** [tony-fav/tasmota-blerry#43] [Q&A] [answered]
Hi guys need some help, I got a ESP-C3-32S 4MB which firmware of tasmota should I flash it with?

**Comment by tony-fav** [comment, marked as answer]
For an ESP32 C3,  you should use https://github.com/tasmota/install/blob/main/firmware/unofficial/tasmota32c3-bluetooth.bin

**Comment by 0allenc0** [comment]
The above link now returns a 404
Does anyone know where to find historical C3 firmware with bluetooth support for the non-safeboot Tasmota? 
I have a Switchbot Plug Mini with 11.1 and trying to figure out how to get BLE support on it
